test/infrastructure/docker/api/v1beta1/dockermachine_types.go

```diff
@@ -30,6 +30,13 @@ const (
 
 // DockerMachineSpec defines the desired state of DockerMachine.
 type DockerMachineSpec struct {
+	// InstanceName indicates the name of the Docker container associated with this DockerMachine.
+	// Since the provider ID is not set until after the container is online, this field is used to
+	// maintain the association. If it is not populated, the name of the container will be the same
+	// as the name of the owner Machine.
+	// +optional
+	InstanceName string `json:"instanceName,omitempty"`
```

Comment:
I'm trying to solve a problem related to [this line](https://github.com/kubernetes-sigs/cluster-api/blob/1acf00d8ab03acd9e627bb4a596bc09d2121547d/test/infrastructure/docker/internal/controllers/dockermachine_controller.go#L155). Instead of pointing `docker.NewMachine()` at `machine.Name`, I'd like to point it at an existing container created by the NodePool since in this case the parent Machine doesn't exist yet. I'm not really sure how else the DockerMachine controller would be able to know the name of the existing `docker.Machine`. Do you have any suggestions?

Reply:
I've refactored it out to use the DockerMachine name instead, I think we're good here.